src/Microsoft.AspNet.OData.Shared/Common/ODataPathHelper.cs

```diff
@@ -0,0 +1,71 @@
+//-----------------------------------------------------------------------------
+// <copyright file="ODataPathHelper.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData.Common
+{
+    /// <summary>
+    /// Helper methods for <see cref="ODataPath"/>.
+    /// </summary>
+    public static class ODataPathHelper
+    {
+        /// <summary>
+        /// Get the keys from a <see cref="KeySegment"/>.
+        /// </summary>
+        /// <param name="keySegment">The <see cref="KeySegment"/> to extract the keys.</param>
+        /// <returns>Dictionary of keys.</returns>
+        public static Dictionary<string, object> GetKeysFromKeySegment(KeySegment keySegment)
+        {
+            if (keySegment == null)
+            {
+                throw Error.ArgumentNull(nameof(keySegment));
+            }
+
+            Dictionary<string, object> keys = new Dictionary<string, object>();
+
+            foreach (KeyValuePair<string, object> kvp in keySegment.Keys)
+            {
+                keys.Add(kvp.Key, kvp.Value);
+            }
+
+            return keys;
+        }
+
+        /// <summary>
+        /// Get the position of the next <see cref="KeySegment"/> in a list of <see cref="ODataPathSegment"/>.
+        /// </summary>
+        /// <param name="pathSegments">List of <see cref="ODataPathSegment"/>.</param>
+        /// <param name="currentPosition">Current position in the list of <see cref="ODataPathSegment"/>.</param>
+        /// <returns>Position of the next <see cref="KeySegment"/>.</returns>
```

Comment:
You should also state that -1 is returned if the segment at the target position is not a key segment or if an out of range value is passed for `currentPosition`